### PR TITLE
chore: upgrade to polkadot v0.9.28

### PR DESCRIPTION
**Pull request by Roznovjak**
**Notable changes:**
Rename: `Class` -> `Collection`, `Instance` -> `Item`
Use our own copy of `orml_currencies` pallet from `warehouse`
New benchmarking commands (`node/src/command.rs`)
No need for `#[transactional]` macro
Implemented migration for `pallet_nft` in warehouse
Implemented migration for `pallet_marketplace`
Added migration for `pallet_uniques`
The version of `fixed` crate fixed to `1.17.0`
Use `polkadot` with patched XCM executor
Add `cumulus_pallet_xcmp_queue` to benchmarking

New or updated runtime configuration parameters:
`orml_tokens::MaxReserves = 50`
`orml_tokens::ReserveIdentifier = ()`
`pallet_currencies::CheckAssociatedRelayNumber = RelayNumberStrictlyIncreases`
`pallet_elections_phragmen::MaxCandidated = 1_000`
`pallet_elections_phragmen::MaxVoters = 10_000`
`pallet_uniques::CreateOrigin = NeverEnsureOrigin`
`pallet_uniques::Locker = ()`
`cumulus_pallet_xcmp_queue::WeightInfo = ()`
`orml_xtokens::MultiLocationsFilter = Everyting`
`orml_xtokens::ReserveProvider = AbsoluteReserveProvider`
`orml_xtokens::MinXcmFee = ParachainMinFee`

### CODE REVIEW

File: .github/workflows/workflow.yml
Comment (apopiak) [inline .github/workflows/workflow.yml:12]: should maybe be 0.9.28?

**Review by dmoka — COMMENTED** [review]
Apart from some small details, LGTM. 

It would be nice if someone else could also look at this PR before merging back to master.

File: runtime/common/src/weights/currencies.rs
Comment (dmoka) [inline runtime/common/src/weights/currencies.rs:44]: line 34, `pallet_currencies` instead of `orml_currencies`

(It is frustrating that in 2022, in the most widely used collaboration tool we are not able to comment on collapsed lines. Or am I missing something? :D)

File: pallets/marketplace/src/migration.rs
Comment (Roznovjak) [reply pallets/marketplace/src/migration.rs:129]: With additional print statements. I created a local testnet using `polkadot-launch` (with old runtime version), filled the storage with some data and run this command `cargo run --release --features=try-runtime try-runtime on-runtime-upgrade live --uri ws://localhost:9988` in another directory where I built a node with this update (with `--features try-runtime` flag). The command executes the migrations and provides some additional info. This way I tested that the migration works as expected. You can also point it to live Basilisk chain (`--uri wss://rpc.basilisk.cloud:443`).

File: runtime/common/src/weights/currencies.rs
Comment (Roznovjak) [reply runtime/common/src/weights/currencies.rs:44]: Yes, I know, but I intentionally didn't rename it because it is auto-generated output. But we can rename it, no problem. Re; commenting on collapsed lines, I think it makes sense, because it's not clear what are you intentions. You might want to add the same comment to each line, or just at the begging/end.

File: pallets/marketplace/src/migration.rs
Comment (dmoka) [reply pallets/marketplace/src/migration.rs:129]: that's awesome, good enough!

**Comment by apopiak** [comment]
> `XcmConfig::WeightInfo = ()`
So we didn't add benchmarks for XCM?

**Comment by Roznovjak** [comment]
> > `XcmConfig::WeightInfo = ()`
> > So we didn't add benchmarks for XCM?

Where did you find this line? I can't find it. XcmConfig doesn't have this field.

**Comment by apopiak** [comment]
> > > `XcmConfig::WeightInfo = ()`
> > > So we didn't add benchmarks for XCM?
> 
> Where did you find this line? I can't find it. XcmConfig doesn't have this field.

From your PR description up above :-D

**Comment by Roznovjak** [comment]
> > > > `XcmConfig::WeightInfo = ()`
> > > > So we didn't add benchmarks for XCM?
> > 
> > 
> > Where did you find this line? I can't find it. XcmConfig doesn't have this field.
> 
> From your PR description up above :-D Edit: Looks like you might have meant the `cumulus_pallet_xcmp_queue`?

Yes, you are right. I will update the description. And yes, we don't have benchmarks for it. @enthusiastmartin what to do if we want to add benchmarks for a pallet that we haven't benchmarked before? Is it enough to just add new file with the default weights to the `common_runtime::weights`?

File: Cargo.toml
Comment (apopiak) [reply Cargo.toml:148]: Well this is patching, not dependency inclusion, so you're making decisions for the whole dependency tree.

File: runtime/basilisk/src/lib.rs
Comment (apopiak) [inline runtime/basilisk/src/lib.rs:442]: I'm confused, I saw the events being removed from some tests, no?

File: pallets/duster/src/tests.rs
Comment (apopiak) [inline pallets/duster/src/tests.rs:150]: There is a comment in the runtime that orml_currencies was forked to keep these events, but they seem to be removed here?

File: runtime/basilisk/src/lib.rs
Comment (apopiak) [inline runtime/basilisk/src/lib.rs:838]: let's add a comment explaining that it is locked down? (that's what I read here)

File: runtime/basilisk/src/migrations.rs
Comment (apopiak) [inline runtime/basilisk/src/migrations.rs:43]: Have we tested that this fits in a block with the production data?

**Comment by apopiak** [comment]
> What to do if we want to add benchmarks for a pallet that we haven't benchmarked before?

They need to be added in the benchmark listing in the runtime lib.rs, no?

File: Cargo.toml
Comment (Roznovjak) [reply Cargo.toml:204]: Don't know what was the reason. Removed.

File: node/src/cli.rs
Comment (Roznovjak) [reply node/src/cli.rs:43]: I couldn't figure out how to do it.

File: pallets/duster/src/tests.rs
Comment (Roznovjak) [reply pallets/duster/src/tests.rs:150]: I removed orml events that followed events from the `balances` or `tokens` pallets. So we test that transfers are executed.

File: node/src/cli.rs
Comment (apopiak) [reply node/src/cli.rs:43]: Huh, Maybe ValueEnum doesn't have a derive :shrug:

File: runtime/basilisk/src/migrations.rs
Comment (Roznovjak) [reply runtime/basilisk/src/migrations.rs:43]: Could you please have a look at this [commit](https://github.com/galacticcouncil/Basilisk-node/pull/564/commits/574268219097ba23748a6ba38e81001e02490896)?

**Review by enthusiastmartin — APPROVED** [review]

**Review by enthusiastmartin — APPROVED** [review]
resolve all comments first.

File: pallets/marketplace/src/lib.rs
Comment (Roznovjak) [reply pallets/marketplace/src/lib.rs:99]: After a discussion we decided to use the migration.

File: runtime/basilisk/src/migrations.rs
Comment (apopiak) [reply runtime/basilisk/src/migrations.rs:43]: So we're planning to run a test after merging? Or you have run with try-runtime and production data already and it's fine?

File: pallets/duster/src/tests.rs
Comment (apopiak) [reply pallets/duster/src/tests.rs:150]: Ah, so `pallet_currencies` keeps *some* but not all events?

File: pallets/exchange/src/lib.rs
Comment (apopiak) [reply pallets/exchange/src/lib.rs:96]: Doesn't need to block merging this PR I guess as it's a different concern, but we should have a task tracking this IMO

File: pallets/duster/src/tests.rs
Comment (Roznovjak) [reply pallets/duster/src/tests.rs:150]: It keeps all events, but I removed events from tests if there was an event from `balances` or `tokens` corresponding to the same transfer. A transfer emits two events, one from `balances/tokens` and one from `currencies`

File: runtime/basilisk/src/migrations.rs
Comment (Roznovjak) [reply runtime/basilisk/src/migrations.rs:43]: I tested it with try-runtime feature against the production data and it consumed less than 2% of the total block weight.

**Pull request merged**
